fem/restriction.cpp

```diff
@@ -48,15 +48,16 @@ ElementRestriction::ElementRestriction(const FiniteElementSpace &f,
       for (int e = 0; e < ne; ++e)
       {
          const FiniteElement *fe = fes.GetFE(e);
-         const TensorBasisElement* el =
-            dynamic_cast<const TensorBasisElement*>(fe);
-         if (el) { continue; }
+         auto el_t = dynamic_cast<const TensorBasisElement*>(fe);
+         auto el_n = dynamic_cast<const NodalFiniteElement*>(fe);
+         if (el_t || el_n) { continue; }
          MFEM_ABORT("Finite element not suitable for lexicographic ordering");
       }
       const FiniteElement *fe = fes.GetTypicalFE();
-      const TensorBasisElement* el =
-         dynamic_cast<const TensorBasisElement*>(fe);
-      const Array<int> &fe_dof_map = el->GetDofMap();
+      auto el_t = dynamic_cast<const TensorBasisElement*>(fe);
+      auto el_n = dynamic_cast<const NodalFiniteElement*>(fe);
+      const Array<int> &fe_dof_map =
+         (el_t) ? el_t->GetDofMap() : el_n->GetLexicographicOrdering();
```

Comment:
I assume this was needed for non-tensor-product mesh. Where exactly do we need lexicographic ordering in the element restriction for such meshes?

Reply:
I don't think we have any kernels that process such data layout?

Reply:
I use it in `GetPeriodicPositions` in tmop_tools.hpp to go from H1 to L2 assuming same order spaces with Gauss-Lobatto support points. With the added code, the transition works for both tensor and non-tensor meshes.

Correct, the PA kernels won't work with that, they abort for non-tensor.

The transition can be implemented in other ways, but I wanted to reuse existing code.